### PR TITLE
move patternt into header file

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- n/a Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 830fdc003a5fdd0d8d946e59a139c53b910801a6).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100282477

File: jbmc/src/java_bytecode/pattern.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*******************************************************************\
+
+Module:
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
```
Comment (peterschrammel) [inline jbmc/src/java_bytecode/pattern.h:8]: doxygen `\file comment...` missing

File: jbmc/src/java_bytecode/pattern.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*******************************************************************\
+
+Module:
```
Comment (peterschrammel) [inline jbmc/src/java_bytecode/pattern.h]: module missing

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 4b05108f01b6da499e717398f24ef0eacf198ff3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100299085

**Pull request merged**